autorest/azure/token.go

```diff
@@ -303,15 +303,12 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	var newToken Token
 	var aadError AADServiceError
 	err = autorest.Respond(resp,
-		autorest.WithErrorUnlessOK(),
-		autorest.ByUnmarshallingSuccessandErrorJSON(&newToken, &aadError),
+		autorest.WithServiceErrorUnlessOK(&newToken, &aadError),
```

Comment:
@jkapil You should not have to pass in the `aadError`. `WithServiceErrorUnlessOK` can just return it as long as it "inherits" from `error`. I believe we do this elsewhere.